packages/ui/src/components/Button/SocialIconButton.tsx

```diff
@@ -0,0 +1,28 @@
+import { ConnectorMetadata } from '@logto/schemas';
+import classNames from 'classnames';
+import React from 'react';
+
+import styles from './SocialIconButton.module.scss';
+
+type Props = {
+  className?: string;
+  connector: Pick<ConnectorMetadata, 'id' | 'logo'>;
+  onClick?: (id: string) => void;
+};
+
+const SocialIconButton = ({ className, connector, onClick }: Props) => {
+  const { id, logo } = connector;
+
+  return (
+    <button
+      className={classNames(styles.socialButton, className)}
+      onClick={() => {
+        onClick?.(id);
```

Comment:
Since the `id` also comes from the parent component, we don't really need to pass it back